platform/search/bundle.json

```diff
@@ -0,0 +1,27 @@
+{
+    "name": "Search",
+    "description": "Allows the user to search through the file tree.",
+    "extensions": {
+        "components": [
+            {
+                "provides": "searchService",
+                "type": "provider",
+                "implementation": "GenericSearchProvider.js",
+                "depends": [ "$q", "objectService", "workerService", "roots[]" ]
```

Comment:
Instead of depending on `roots[]` here, I think you could depend on a constant `GENERIC_SEARCH_ROOTS` which would be an array of identifiers to serve as the starting points for indexing.

In this same bundle definition, you could then also provide a default value for this:

```
"constants": [
    {
        "key": "GENERIC_SEARCH_ROOTS",
        "value": [ "ROOT" ],
        "priority": "fallback"
    }
]
```

This would mean that GenericSearchProvider would traverse the whole tree in its default configuration, but would be open for reconfiguration. (For instance, if at deployment time you know that most of your stuff is going to be in ElasticSearch except for a telemetry, you could add a configuration bundle defining `GENERIC_SEARCH_ROOTS` to include just the name of that telemetry dictionary.)